cmd/notation/internal/truststore/truststore.go

```diff
@@ -37,17 +37,23 @@ func AddCert(path, storeType, namedStore string, display bool) error {
 		return errors.New("named store name needs to follow [a-zA-Z0-9_.-]+ format")
 	}
 
-	// check if the target path is a cert (support PEM and DER formats)
-	if _, err := corex509.ReadCertificateFile(certPath); err != nil {
+	// check if the target path is a x509 certificate
+	// (support PEM and DER formats)
+	certs, err := corex509.ReadCertificateFile(certPath)
+	if err != nil {
 		return err
 	}
+	if len(certs) == 0 {
+		return errors.New("no valid certificate found in the file")
+	}
```

Comment:
1. can you please add unit test for this? Also should we add this check to `ReadCertificateFile` method ?
3. s/`no valid certificate found in the "empty file"

Reply:
> 1. can you please add unit test for this? Also should we add this check to `ReadCertificateFile` method ?
> 2. s/`no valid certificate found in the "empty file"

1. hmm, so both notation CLI and notation-go are using this function. But they have different behaviors and returns different errors when encounter this situation. For the `notation certificate list` command, actually no error is returned, the file is just skipped. So probably we could just add this quick fix in CLI?
2. Code updated.